# GiniQC

#### Dependencies

- Python 3.6
- Cooler (`pip install cooler`)
- numpy (`pip install numpy`)
- pandas (`pip install pandas`)

#### Usage

```
Usage: bash GiniQC.sh [-h] -f FILE(s) -o OUTFILE -b BEDFILE [-c CISTHRESHOLD] [-g GINITHRESHOLD] [-r READSTHRESHOLD] [-a MAXABERRATION]
	-h|--help				prints this message
	-f FILE(s)			path to cooler matrix file (must end in .cool) or path to a list of files (any other extension)
	-o OUTFILE			desired name for output files, including pairs file
	-b BEDFILE			paired-end fastq files corresponding to a single cell
	-c CISTHRESHOLD		user-defined value (default: 80% cis)
	-g GINITHRESHOLD	minimum GiniQC value (if unspecified, our tool will suggest a threshold)
	-r READSTHRESHOLD 	minimum number of reads per cell (default: 10,000 reads)
	-a MAXABERRATION 	maximum fold-change in coverage  (default: 2-fold)
```

When you would like to run GiniQC on several files, please note that the list of files must be in the same directory as the specified files OR the list of files must specify the full path of the files.

#### Sample run

To test the code using a single file, the user can try the following command:
```
bash GiniQC.sh -f test/cell1.chrom.cool -o cell1_stats.txt -b helper_files/mm10.chroms.bed
```

To test the code using a list of files, the user can try the following command:
```
bash GiniQC.sh -f test/file_list.txt -o test_output.txt -b helper_files/mm10.chroms.bed
```

#### References

The raw data used to create the contact matrices provided in this repository as a test dataset were generated by Stevens et al. (2017):

Stevens, T.J., Lando, D., Basu, S., Atkinson, L.P., Cao, Y., Lee, S.F., Leeb, M., Wohlfahrt, K.J., Boucher, W., O’Shaughnessy-Kirwan, A., et al. (2017). 3D structures of individual mammalian genomes studied by single-cell Hi-C. Nature 544, 59–64.
